File: output/CodeGen/avatar/Java/Python/atcoder_ABC109_B.py
class atcoder_ABC109_B {
    def main(args):
        sc = Scanner(sys.stdin)
        N = sc.next()
        map = {}
        for i in range(N):
            map[i] = sc.next()
        ans = True
        past = {}
        next = ""
        for i in range(N):
            if past.get(i) is not None:
                ans = False
                break
            past[i] = map[i]
            if i!= 0:
                if next!= map[i][0:1]:
                    ans = False
                    break
                next = map[i][-1]
        if ans:
            print("Yes")
        else:
            print("No")

}

File: output/CodeLlama/avatar/Python/Java/atcoder_ABC109_B.java
import java.util.*;
public class atcoder_ABC109_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] l = new String[n];
        for (int i = 0; i < n; i++) {
            l[i] = sc.next();
        }
        boolean f = true;
        for (int i = 0; i < n - 1; i++) {
            if (l[i].charAt(l[i].length() - 1) != l[i + 1].charAt(0)) {
                f = false;
                break;
            }
        }
        System.out.println(f ? "Yes" : "No");
    }
}